docs/adrs/adr-0001-dependency_management.md

```diff
@@ -0,0 +1,40 @@
+# Tooling for Better Dependency Management
+
+* Status: 
+* Deciders: 
+* Date: 2024-03-08
+
+
+## Context and Problem Statement
+Currently dependencies are very loosly defined in the `setup.py`. This can lead to issues when installing the package, an example of which is detailes [in this issue](https://github.com/mozilla/mozanalysis/issues/202).  We need to find a way to more clearly define dependencies in a way that will not be disruptive for developers and users.
+
+## Decision Drivers
+- Minimize the potential for incompatible dependencies
+- Have guardrails/automation so that changes to dependencies made by developers will easily be reflected in the PRs they make
+- Ideally no disruption to user workflow
+- Minimal distruption to developer workflow
+
+## Options
+
+### 1. pip-tools to generate requirements.txt
+Jetstream currently uses this process to manage dependencies.  It involves uses pip-tools to generate a requirements.txt and requirements.in file from the setup.py. [dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/) is a built-in github tool that is used to periodically check for dependency updates and create pull requests to implement them.  [Here](https://github.com/mozilla/jetstream/pull/2017) is an example of such a pull request.
+
+* +: Jetstream uses it already so there is an existing base of experties in the data team
+* +: Dependabot is a built-in github tool that has additional uses like checking for security vulnerabilities
+* -: Still using pip's dependency resolver, which is fast but can have issues with incompatible versions.  This may be mitigated or even solved by the way pip-tools pins dependencies.
+* -: Jetstream doesn't use `pyproject.toml` for configuring dependencies, though it does seem to be supported by `pip-tools`
```

Comment:
Yea I think this is more a relic of Jetstream's history than an intentional or necessary decision, and we could change to `pyproject.toml` if we want. Worth calling out here though.

Reply:
Based on some online sleuthing it seems like the `pyproject.toml` support is relatively new too